node/src/indexer/tee.rs

```diff
@@ -0,0 +1,63 @@
+use std::sync::Arc;
+
+use mpc_contract::tee::proposal::AllowedDockerImageHash;
+use tokio::sync::watch;
+
+use crate::indexer::{
+    lib::{get_mpc_tee_state, wait_for_full_sync},
+    IndexerState,
+};
+
+/// This future waits for the indexer to fully sync, and returns
+/// a [`watch::Receiver`] that will be continuously updated with the latest
+/// allowed [`AllowedDockerImageHash`]es when a change is detected
+/// on the MPC smart contract.
+pub async fn monitor_allowed_docker_images(
+    indexer_state: Arc<IndexerState>,
+) -> watch::Receiver<Vec<AllowedDockerImageHash>> {
+    const CONTRACT_STATE_REFRESH_INTERVAL: std::time::Duration = std::time::Duration::from_secs(1);
+
+    tracing::debug!(target: "indexer", "awaiting full sync to read mpc contract state");
+    wait_for_full_sync(&indexer_state.client).await;
+
+    let fetch_allowed_image_hashes = {
+        async move || loop {
+            match get_mpc_tee_state(
+                indexer_state.mpc_contract_id.clone(),
+                &indexer_state.view_client,
+            )
+            .await
+            {
+                Ok((block_height, mut allowed_images_on_contract)) => {
+                    let allowed_images: Vec<AllowedDockerImageHash> =
+                        allowed_images_on_contract.get(block_height);
+                    break allowed_images;
+                }
+                Err(e) => {
+                    tracing::error!(target: "mpc", "error reading tee state from chain: {:?}", e);
+                    continue;
```

Comment:
Maybe add a progressive timeout here? 
Repeatedly failing here means there is something wrong and will eventually lead to the node getting kicked out.

Hence, it could be valuable to somehow get the attention of the node operators.

Not sure if/how we are monitoring error messages, but maybe one way to do it is to expose a metric, measuring the number of failed calls to this endpoint. The metric can then easily be integrated into a dashboard.

Reply:
Perhaps worth considering an exponential backoff here https://crates.io/crates/backoff.